### PR TITLE
Nf/updated bigfix components and fetch errors handling

### PR DESCRIPTION
**Pull request by NabilFurmoli**

### CODE REVIEW

File: Keas.Mvc/Services/BigfixService.cs [hunk 1]
```diff
@@ -66,12 +66,19 @@ public async Task<BigfixComputerSearchResult[]> GetComputersByName(string name)
...
-            using (var bf = GetClient())
+            try {
+                using (var bf = GetClient())
+                {
+                    var results = await bf.Computers.Get(id);
+
+                    return new BigFixComputerProperties(results);
+                }
+            } 
+            catch (Exception)
```
Comment (srkirkland) [inline Keas.Mvc/Services/BigfixService.cs]: Do no swallow exceptions inside of a service.  Other callers may also use this service and should have the option of catching specific errors

File: Keas.Mvc/Controllers/Api/EquipmentController.cs [hunk 2]
```diff
@@ -324,16 +324,24 @@ public async Task<IActionResult> GetHistory(int id)
             return Json(history);
         }
 
-        public async Task<BigFixComputerProperties> GetComputer(string id)
+        public async Task<IActionResult> GetComputer(string id)
         {
-            return await this._bigfixService.GetComputer(id);
+            var result = await this._bigfixService.GetComputer(id);
```
Comment (srkirkland) [inline Keas.Mvc/Controllers/Api/EquipmentController.cs]: if you want to try/catch here, that would be a better spot.  Also, why not just handle it in the client?  is the appropriate data not being transferred through?

File: Keas.Mvc/Services/BigfixService.cs (hunk 1 above)
Comment (NabilFurmoli) [reply Keas.Mvc/Services/BigfixService.cs]: fixed

File: Keas.Mvc/Controllers/Api/EquipmentController.cs (hunk 2 above)
Comment (NabilFurmoli) [reply Keas.Mvc/Controllers/Api/EquipmentController.cs]: so, as you mentioned, for now, I handled the exception here. one thing I am not quite sure is that how bigfix server handles the invalid id. if we don't catch the exception here, on our client-side we receive a 500 Internal Server Error.

**Comment by NabilFurmoli** [comment]
@jSylvestre the PR does not add new features to the application, in this PR I have only reconstructed Bigfix related components to make them more readable and cleaner, also I have updated error handling logics used in those components for fetching data.

File: Keas.Mvc/ClientApp/components/Equipment/EquipmentBigFixInfo.tsx [hunk 3]
```diff
@@ -127,27 +110,34 @@ export default class EquipmentBigFixInfo extends React.Component<IProps, IState>
...
-        if (!response.ok) {
-            // show the invalid Bigfix-id message.
+        let response = null;
+        try {
+            response = await this.context.fetch(
+                `/api/${this.context.team.slug}/equipment/GetComputer/${id}`
+            );
+        } catch (err) {
+            if (err.message === "Not Found") {
+                toast.error("Not a valid Bigfix id, please make sure to enter a valid Bigfix id.");
```
Comment (laholstege) [inline Keas.Mvc/ClientApp/components/Equipment/EquipmentBigFixInfo.tsx]: @NabilFurmoli this shouldn't be a toast, the modal should pop up with an error telling the user that their id is invalid.

File: Keas.Mvc/ClientApp/components/Equipment/EquipmentBigFixSearchId.tsx [hunk 4]
```diff
@@ -13,13 +16,16 @@ import {
 } from 'reactstrap';
 import { Context } from '../../Context';
 import { IBigFixSearchedName } from '../../Types';
+import { threadId } from 'worker_threads';
+import { isThisQuarter } from 'date-fns';
```
Comment (srkirkland) [inline Keas.Mvc/ClientApp/components/Equipment/EquipmentBigFixSearchId.tsx]: @NabilFurmoli remove these lines, they are not used anywhere

File: Keas.Mvc/Controllers/Api/EquipmentController.cs [hunk 5]
```diff
@@ -324,16 +325,41 @@ public async Task<IActionResult> GetHistory(int id)
...
+                 return Json(result);
+            }
+            catch (BigfixApiException ex) 
+            {
+                if (ex.StatusCode == System.Net.HttpStatusCode.NotFound)
+                {
+                    return NotFound();
+                }
+                else
+                    throw;
```
Comment (srkirkland) [inline Keas.Mvc/Controllers/Api/EquipmentController.cs]: keep using `{ }` for else statements even if they are one-liners

File: Keas.Mvc/Controllers/Api/EquipmentController.cs [hunk 6]
```diff
@@ -324,16 +325,41 @@ public async Task<IActionResult> GetHistory(int id)
...
+                {
+                    var result = await this._bigfixService.GetComputersByName(value);
+                    if (result.Length == 0)
+                        return NotFound();
+
+                    else
+                        return Json(result);
+
+                } catch (BigfixApiException) {
+                    throw;
```
Comment (srkirkland) [inline Keas.Mvc/Controllers/Api/EquipmentController.cs]: @NabilFurmoli don't bother catching if you are just going to re-throw

File: Keas.Mvc/ClientApp/components/Equipment/EquipmentBigFixSearchId.tsx [hunk 7]
```diff
@@ -219,45 +233,61 @@ export default class EquipmentBigFixSearchId extends React.Component<
...
-        isValidSearch: false
-      });
-    } else {
-      const firstFiveName = response.slice(0, 5);
-      this.setState({
-        isFetched: true,
-        isSearching: false,
-        listOfComputers: firstFiveName
-      });
+      debugger;
```
Comment (laholstege) [inline Keas.Mvc/ClientApp/components/Equipment/EquipmentBigFixSearchId.tsx]: @NabilFurmoli you should take this out, and fix the other tslint warnings on this page. but other than that looks good! :shipit:

**Pull request merged**